core/capabilities/compute/compute.go

```
@@ -197,7 +197,10 @@ func (c *Compute) initModule(id string, cfg *host.ModuleConfig, binary []byte, r
 	computeWASMInit.WithLabelValues(requestMetadata.WorkflowID, requestMetadata.ReferenceID).Observe(float64(initDuration))
 
 	m := &module{module: mod}
-	c.modules.add(id, m)
+	err = c.modules.add(id, m)
+	if err != nil {
+		c.log.Errorf("failed to add module to cache: %s", err.Error())
```

why do we not want to return the error if there's an ID collision?

---

we were talking about this on [this comment](https://github.com/smartcontractkit/chainlink/pull/15901#discussion_r1913217557)

---

@agparadiso We shouldn't log this as an error, warning is more appropriate (since this is actually expected/desirable behaviour)